Decisions.Microsoft365.Common/API/Calendar/Microsoft365Event.cs

```diff
@@ -0,0 +1,275 @@
+        public bool? ResponseRequested { get; set; }
+
+        [WritableValue]
+        [JsonProperty("seriesMasterId")]
+        public string? SeriesMasterId { get; set; }
+
+        [WritableValue]
+        [JsonProperty("showAs")]
+        public string? ShowAs { get; set; }
+
+        [WritableValue]
+        [JsonProperty("type")]
+        public string? Type { get; set; }
+
+        [WritableValue]
+        [JsonProperty("webLink")]
+        public string? WebLink { get; set; }
+
+        [WritableValue]
+        [JsonProperty("onlineMeetingUrl")]
+        public string? OnlineMeetingUrl { get; set; }
+
+        [WritableValue]
+        [JsonProperty("isOnlineMeeting")]
+        public bool? IsOnlineMeeting { get; set; }
+
+        [WritableValue]
+        [JsonProperty("onlineMeetingProvider")]
+        public string? OnlineMeetingProvider { get; set; }
+
+        [WritableValue]
+        [JsonProperty("allowNewTimeProposals")]
+        public bool? AllowNewTimeProposals { get; set; }
+
+        [WritableValue]
+        [JsonProperty("occurrenceId")]
+        public string? OccurrenceId { get; set; }
+
+        [WritableValue]
+        [JsonProperty("isDraft")]
+        public bool? IsDraft { get; set; }
+
+        [WritableValue]
+        [JsonProperty("hideAttendees")]
+        public bool? HideAttendees { get; set; }
+
+        [WritableValue]
+        [JsonProperty("responseStatus")]
+        public Microsoft365EventStatus? ResponseStatus { get; set; }
+
+        [WritableValue]
+        [JsonProperty("body")]
+        public Microsoft365EventBody? Body { get; set; }
+
+        [WritableValue]
+        [JsonProperty("start")]
+        public Microsoft365EventTime? Start { get; set; }
+
+        [WritableValue]
+        [JsonProperty("end")]
+        public Microsoft365EventTime? End { get; set; }
+
+        [WritableValue]
+        [JsonProperty("location")]
+        public Microsoft365LocationName? Location { get; set; }
+
+        [WritableValue]
+        [JsonProperty("locations")]
+        public Microsoft365LocationName[]? Locations { get; set; }
+
+        [WritableValue]
+        [JsonProperty("recurrence")]
+        public string? Recurrence { get; set; }
+
+        [WritableValue]
+        [JsonProperty("attendees")]
+        public Microsoft365EventAttendee[]? Attendees { get; set; }
+
+        [WritableValue]
+        [JsonProperty("organizer")]
+        public Microsoft365EventOrganizer? Organizer { get; set; }
+
+        [WritableValue]
+        [JsonProperty("onlineMeeting")]
+        public string? OnlineMeeting { get; set; }
+        
+        public static Microsoft365Event? JsonDeserialize(string content)
+        {
+            try
+            {
+                return JsonConvert.DeserializeObject<Microsoft365Event>(content);
+            }
+            catch (Exception ex)
+            {
+                throw new BusinessRuleException("Could not deserialize result.", ex);
```

Comment:
Does this Log also? We may want to use LoggedException and maybe a Debug message that include the content of the response?

Ideally we want to be able to understand where a problem is happening just based on information from the customer, incase we are unable to recreate it.
 
No change required, but something to think about as you go.